packages/replay/src/eventBuffer/EventBufferArray.ts

```diff
@@ -0,0 +1,75 @@
+import type { ReplayRecordingData } from '@sentry/types';
+
+import type { EventBuffer, RecordingEvent } from '../types';
+
+interface EventsGroup {
+  checkoutTimestamp: number;
+  events: RecordingEvent[];
+}
+
+/**
+ * A basic event buffer that does not do any compression.
+ * Used as fallback if the compression worker cannot be loaded or is disabled.
+ */
+export class EventBufferArray implements EventBuffer {
+  private _events: EventsGroup[];
+
+  public constructor() {
+    this._events = [];
+  }
+
+  /** @inheritdoc */
+  public get pendingLength(): number {
+    return this.pendingEvents.length;
+  }
+
+  /** @inheritdoc */
+  public get pendingEvents(): RecordingEvent[] {
+    return this._events.reduce((acc, { events }) => [...events, ...acc], [] as RecordingEvent[]);
+  }
+
+  /** @inheritdoc */
+  public getFirstCheckoutTimestamp(): number | null {
+    return (this._events[0] && this._events[0].checkoutTimestamp) || null;
```

Comment:
`0` will result in `null`? better use `??` instead

```suggestion
    return this._events.at(0)?.checkoutTimestamp ?? null;
```

Reply:
We avoid `??` because it is transpiled to a rather verbose form. In this case, a timestamp of `0` is not really relevant for us, because we're unlikely to have a timestamp in 1970 ;) So I'd say this is fine in this case!